### PR TITLE
Kubernetes monitor improvements

### PR DESCRIPTION
**Pull request by Kami**
This pull request includes the following changes:

1. Make sure we log any errors we receive from Kubernetes API under warn log level and also include actual response body. This should make it easier to troubleshoot various configuration and permission related issue - as suggested in #631.

2. While trying to reproduce #631, I encountered weird behavior when trying to stop the agent before the Kubernetes monitor has been fully initialized (e.g. if we receive an error from the API or we fail to connect to it).

In such scenario, CTRL+C would have no effect - the reason for that is that the "stop" signal was not correctly propagated to all the underlying objects and threads.

My change is probably not 100% ideal, but based on my testing, it works. The problem is that the current code is quite convoluted so doing any other major refactoring would be more risky.

I'm also looking into adding a test case for it, but it's not totally straight forward.

## TODO

- [x] Tests for stop process edge case fix

### CODE REVIEW

File: scalyr_agent/builtin_monitors/kubernetes_monitor.py [hunk 1]
```diff
@@ -2734,12 +2748,16 @@ def start(self):
...
     def stop(self, wait_on_join=True, join_timeout=5):
+        self.__stopped = True
+
         if self.__controlled_warmer is not None:
             self.__controlled_warmer.stop(
                 wait_on_join=wait_on_join, join_timeout=join_timeout
             )
 
-        self.__thread.stop(wait_on_join=wait_on_join, join_timeout=join_timeout)
+        if self.__thread.isAlive():
```
Comment (Kami) [inline scalyr_agent/builtin_monitors/kubernetes_monitor.py:2758]: This is needed because threads might not have been started yet when stop is called.

```bash
2020-10-02 14:00:26.258Z ERROR [core] [scalyr_logging.py:618] Failed to stop the metric log due to an exception :stack_trace:
  Traceback (most recent call last):
    File "/home/kami/w/scalyr/scalyr-agent-2/scalyr_agent/monitors_manager.py", line 201, in stop_manager
      monitor.stop(join_timeout=max_wait)
    File "/home/kami/w/scalyr/scalyr-agent-2/scalyr_agent/builtin_monitors/kubernetes_monitor.py", line 4195, in stop
      self.__container_checker.stop(wait_on_join, join_timeout)
    File "/home/kami/w/scalyr/scalyr-agent-2/scalyr_agent/builtin_monitors/kubernetes_monitor.py", line 2746, in stop
      self.__thread.stop(wait_on_join=wait_on_join, join_timeout=join_timeout)
    File "/home/kami/w/scalyr/scalyr-agent-2/scalyr_agent/util.py", line 1226, in stop
      self.join(join_timeout)
    File "/home/kami/w/scalyr/scalyr-agent-2/scalyr_agent/util.py", line 1248, in join
      threading.Thread.join(self, timeout)
    File "/home/kami/.pythonz/pythons/CPython-3.6.9/lib/python3.6/threading.py", line 1051, in join
      raise RuntimeError("cannot join thread before it is started")
  RuntimeError: cannot join thread before it is started
```

**Review by yanscalyr — APPROVED** [review]
LGTM
